### ISSUE TITLE
serve APR Q4K GPU chat samples with a wall-clock-hash RNG: `seed` does nothing, sampled output is unreproducible (sample_with_temperature)

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#3786]
Found by aprender-a8 in the #3760 sampler sweep (`docs/audits/sampler-sweep-PMAT-3760.md` §D, highest-priority row).

## Defect

`crates/aprender-serve/src/cli/apr_inference.rs:288` `sample_with_temperature(logits, temperature, top_k)` draws its uniform sample from a hash of the wall clock:

```rust
let mut hasher = DefaultHasher::new();
SystemTime::now().hash(&mut hasher);
let r = (hasher.finish() as f32 / u64::MAX as f32) * sum;
```

It is called from `crates/aprender-serve/src/api/apr_q4k_scheduler.rs:303` and `:333`, which is the serve APR Q4K GPU chat path. So on that path a request's `seed` never reaches the sampler. The same request and seed give different output on every call.

It is also a second copy of the top-k filter, next to the shared `realizar::sampling::draw` introduced by #3760. It has its own edge case: `top_k == 0` indexes `top[0]` of an empty slice. Today's callers pass 40.

This is the same class as #3754 and #3760: a flag that does nothing, with no warning.

## done_when

- [ ] `sample_with_temperature` is removed, or delegates to `realizar::sampling::draw_seeded` with a `StdRng` seeded from the request's `seed` (default `sampling::DEFAULT_SEED`). The APR Q4K scheduler threads the seed through.
- [ ] The same request and seed twice give byte-identical output on the APR Q4K GPU chat path. A different seed gives different output. Temperature 0 or `top_k` 1 stays greedy. Each is a test that is RED at 0.69.0.
- [ ] No wall-clock-derived randomness is left in any token-selection site (`git grep -n 'SystemTime::now().hash'` over the generation paths is empty, or each remaining hit is justified).

Refs #3760 #3754 #3710

**Comment by noahgift** [comment]
closed by triage (aprender-dd, #3988 FIXED row, cop OK): fixed by `d9ba90cff`, all in v0.69.1. Evidence: serve APR Q4K GPU honours seed
